### PR TITLE
Scenarios with steps

### PR DESCRIPTION
**Pull request by MaximRaicov**
Finished scenarios with steps:
added scenarios
added css selectors
updated stepDefs

### CODE REVIEW

File: src/main/java/components/HeaderComponent.java
Comment (vdones) [inline src/main/java/components/HeaderComponent.java]: we decided to put elements of this component in the abstractPageObject, aren't we?

File: src/main/java/pages/HomePage.java
Comment (vdones) [inline src/main/java/pages/HomePage.java]: move it below WebElements, not between

File: src/main/java/pages/ShoppingCartPage.java
Comment (vdones) [inline src/main/java/pages/ShoppingCartPage.java]: code convention, variable names begin with little letter, like shoppingCartIsEmpty

File: src/test/java/stepDefinition/HomePageStepDef.java
Comment (vdones) [inline src/test/java/stepDefinition/HomePageStepDef.java]: why you left comments?

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (vdones) [inline src/test/java/stepDefinition/ProductPageStepDef.java]: why you left comments?

File: src/test/resources/features/AddProductToCart.feature
Comment (vdones) [inline src/test/resources/features/AddProductToCart.feature]: you can put the first two steps in the background

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (vdones) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: assertEquals instead of assertTrue

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (vdones) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: add assert message please

**Review by vdones — CHANGES_REQUESTED** [review]

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (CristianManic) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: or you can user **assertThat()** and add log message

File: src/main/java/pages/HomePage.java
Comment (MaximRaicov) [reply src/main/java/pages/HomePage.java]: Done.
Thank You!

File: src/main/java/components/HeaderComponent.java
Comment (MaximRaicov) [reply src/main/java/components/HeaderComponent.java]: Done.
Thank You!

File: src/main/java/pages/ShoppingCartPage.java
Comment (MaximRaicov) [reply src/main/java/pages/ShoppingCartPage.java]: Done.
Thank You!

File: src/test/java/stepDefinition/HomePageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/HomePageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ProductPageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

File: src/test/resources/features/AddProductToCart.feature
Comment (MaximRaicov) [reply src/test/resources/features/AddProductToCart.feature]: Done.
Thank You!

File: src/test/java/cucumber_runner/TestRunner.java
Comment (CristianManic) [inline src/test/java/cucumber_runner/TestRunner.java:15]: why would you need this change?

File: src/test/java/stepDefinition/GenericsStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/GenericsStepDef.java:24]: why waitForPageload is after assert?

File: src/test/java/stepDefinition/CategoryPageStepDef.java
Comment (CristianManic) [inline src/test/java/stepDefinition/CategoryPageStepDef.java]: what is the purpose of this 2 lines?

File: src/test/java/stepDefinition/GenericsStepDef.java
Comment (CristianManic) [inline src/test/java/stepDefinition/GenericsStepDef.java]: can we click on field?

File: src/test/java/stepDefinition/HomePageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/HomePageStepDef.java]: Pay attention to assert message

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/ProductPageStepDef.java:16]: What do you need method parameter 'message'?

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java:19]: the same issue you don't use method parameter 'product', please check the rest of code cause this can be found in other methods to. Why do you use hardcoded parameters?

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Please don't use a such long lines it's not readable at all

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (CristianManic) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: 1. replace **@And** with a related one
2. what is the purpose of line 35?
3. this step could be replaced with a more generic one, take a look on **user populates fields with following values:** step that already exists

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: as I can understand inputQuantity is a global variable, I suggest to use scenarioContext rather then global variable

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (CristianManic) [reply src/test/java/stepDefinition/ProductPageStepDef.java:16]: Why doesn't he use this parameter to check that message corresponds to the expected one

File: src/test/java/cucumber_runner/TestRunner.java
Comment (MaximRaicov) [reply src/test/java/cucumber_runner/TestRunner.java:15]: I just forgot to change it back.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Same problem with assert message

File: src/test/java/stepDefinition/CategoryPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/CategoryPageStepDef.java]: Because I validate subCategory, It's for assert.
Thank You!

File: src/test/resources/features/AddProductToCart.feature
Comment (IulianCimpoes) [inline src/test/resources/features/AddProductToCart.feature]: In my opinion this  should be moved to the Background

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ProductPageStepDef.java:16]: Because I can't navigate to next page if I will not close this 'alert message'

File: src/test/resources/features/AddProductToCart.feature
Comment (IulianCimpoes) [inline src/test/resources/features/AddProductToCart.feature]: And the next 3 steps also should be moved to the background, after all permutations 4 scenarios should remain so you should add one more scenario

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java:19]: Done.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: replace @and with a related one- Changed.
Because I have to clear field before.
Yes, I use it (inputQuantity = itemsQty;) in other method to cheque quantity.

File: src/test/java/stepDefinition/HomePageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/HomePageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ProductPageStepDef.java:16]: Also, this method will be changed with PR from Andrei

File: src/test/resources/features/AddProductToCart.feature
Comment (MaximRaicov) [reply src/test/resources/features/AddProductToCart.feature]: That's my TC1 result.

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

File: src/test/java/stepDefinition/GenericsStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/GenericsStepDef.java:24]: Because sometimes is not enough time to find an element on a new page.

File: src/test/java/stepDefinition/GenericsStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/GenericsStepDef.java]: Yes, we can.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

File: src/test/resources/features/AddProductToCart.feature
Comment (MaximRaicov) [reply src/test/resources/features/AddProductToCart.feature]: I want to upgrade and maybe change my feature file when we get a final decision with methods.
Thank You!

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Done.
Thank You!

**Review by vdones — APPROVED** [review]

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (CristianManic) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: to parse some data other steps use scenario context

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (CristianManic) [reply src/test/java/stepDefinition/ProductPageStepDef.java:16]: 1. question was why do you parse a parameter from step as you are not using it
2. closing alert message should be done in another step or you should rename your method/step as they don't match

File: src/test/java/stepDefinition/ShoppingCartPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ShoppingCartPageStepDef.java]: Changed.
Thank You!

File: src/test/java/stepDefinition/ProductPageStepDef.java
Comment (MaximRaicov) [reply src/test/java/stepDefinition/ProductPageStepDef.java:16]: Done

File: src/test/java/stepDefinition/CategoryPageStepDef.java
Comment (CristianManic) [reply src/test/java/stepDefinition/CategoryPageStepDef.java]: anyway, line 17 and 18 is the duplicated on line 21 and 22

**Review by CristianManic — CHANGES_REQUESTED** [review]
please resolve conflicts, after that we will merge

**Review by amiordan — APPROVED** [review]

**Pull request merged**
